### PR TITLE
Add Python 3.10, 3.11, 3.12 to the testing

### PR DESCRIPTION
**Pull request by cclauss**
https://devguide.python.org/#status-of-python-branches

Also, allow pip's new dependency resolver to work by providing multiple dependencies in the same command.

## Proposed changes

Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce to **CausalML**?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the [CONTRIBUTING](https://github.com/uber/causalml/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla-assistant.io/uber/causalml)
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc. This PR template is adopted from [appium](https://github.com/appium/appium).

### CODE REVIEW

File: .github/workflows/python-test.yaml [hunk 1]
```diff
@@ -10,7 +10,7 @@ jobs:
       # You can use PyPy versions in python-version.
       # For example, pypy2 and pypy3
       matrix:
-        python-version: [3.6, 3.7, 3.8, 3.9]
+        python-version: ["3.7", "3.8", "3.9", "3.10"]
```
Comment (paullo0106) [inline .github/workflows/python-test.yaml:13]: do we keep 3.6?

Comment (cclauss) [reply .github/workflows/python-test.yaml:13]: Did you see the first line of the commit message?

Comment (ppstacy) [reply .github/workflows/python-test.yaml:13]: Any reasons that we want to drop it @cclauss ? I think we probably should keep 3.6 as some of people might still use it, do you mind add it back? Thanks!

**Comment by ppstacy** [comment]
@cclauss the build seems failed for all the versions - do you mind take a look? Thanks!

**Pull request merged**